src/Symfony/Component/Console/Output/ConsoleSectionOutput.php

```diff
@@ -168,6 +168,13 @@ public function addNewLineOfInputSubmit(): void
      */
     protected function doWrite(string $message, bool $newline)
     {
+        // if there is a new line at the end of message, but we do not add a new line, then simulate if without new line and add new line
+        // a lot of code relies on the newline bool to correctly format the output, it avoid a lot of new logic to handle both cases which are roughly the same
+        if (!$newline && str_ends_with($message, \PHP_EOL)) {
```

Comment:
This should avoid the last part of correctly? Can we fix that there instead of this or are there more related parts?
https://github.com/symfony/symfony/blob/16a4681b5b7a20f60c12f0ab1e2d22dbdf4ba50b/src/Symfony/Component/Console/Output/ConsoleSectionOutput.php#L123

Reply:
Certainly, didn't want to too much of other logics, but having this can certainly clean up some logics done afterwards

Reply:
I removed the last part of this if, as it's not needed anymore, thanks for the insight